### PR TITLE
Harmony: FormEditor

### LINKED ISSUE #847: Harmony: FormEditor

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes

- [x] Bug Fix
- [x] Feature

## Related issues

Resolve #847 

## QA Instructions, Screenshots, Recordings
  
Снимок экрана 2024-02-14 в 12 00 58

Снимок экрана 2024-02-14 в 12 25 43

Снимок экрана 2024-02-14 в 12 00 54

Снимок экрана 2024-02-14 в 12 00 49

### CODE REVIEW

File: src/harmony/FormEditor/FormEditor.module.css [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    border-bottom-right-radius: 0;
+}
+
+.Wrapper_brick_center {
+    border-radius: 0;
+}
+
+/* Order of styles on page: Monaco styles, CSS Modules */
+.Wrapper_brick_top
+:global(.monaco-editor):global(.monaco-editor),
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.module.css:64]: selector is not allowed

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Review by awinogradov — CHANGES_REQUESTED** [review]
Do not rename components

File: src/harmony/FormEditor/FormEditor.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+};
+
+export default meta;
+
+export const Default: StoryObj<typeof FormEditor> = {
+    args: {
+        outline: true,
+        placeholder: 'Some placeholder...',
+        style: {
+            width: 500,
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.stories.tsx:20]: U have this style on container

File: src/harmony/FormEditor/FormEditor.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        },
+    },
+    render(props) {
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.stories.tsx:23]: Is not neccessary

File: src/harmony/FormEditor/FormEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+    ) => {
+        const [focused, setFocused] = useState(false);
+        const [contentHeight, setContentHeight] = useState(height);
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const monacoEditorRef = useRef<any>(null);
+        const extraRef = useRef<HTMLDivElement>(null);
+        const [viewValue, setViewValue] = useState<string | undefined>('');
+        const mounted = useMounted();
+        const { loading, files, uploadFiles } = useUpload(onUploadSuccess, onUploadFail, uploadLink);
+        // @ts-ignore
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.tsx:132]: Delete or add comment

File: src/harmony/FormEditor/FormEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+                            <div className={s.UploadButton}>
+                                <input className={s.UploadInput} onChange={onFileInputChange} type="file" multiple />
+
+                                {nullable(
+                                    loading,
+                                    () => (
+                                        <Text
+                                            as="span"
+                                            size="s"
+                                            color="var(--text-secondary)"
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.tsx:334]: Use CSS

File: src/harmony/FormEditor/FormEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+                                    ),
+                                    <>
+                                        <Button
+                                            size="xs"
+                                            view="ghost"
+                                            text={messages?.attachmentsButton || defaultAttachmentsButtonMessage}
+                                            iconLeft={<IconAttachOutline size="xs" />}
+                                            onClick={onUploadLinkClick}
+                                        />
+                                        <Text as="span" size="s" color="var(--text-secondary)">
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.tsx:348]: Use CSS

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/harmony/FormEditor/FormEditor.stories.tsx (hunk 3 above)
Comment (DenisVorop) [reply src/harmony/FormEditor/FormEditor.stories.tsx:22]: added `<Layout />` to render instead `style: { width: 500 }`

File: src/harmony/FormEditor/FormEditor.module.css [hunk 7]
```diff
@@ -0,0 +1,111 @@
+.UploadInput {
+    position: absolute;
+    z-index: 2;
```
[the remark this answers is not in the document]
Comment (DenisVorop) [reply src/harmony/FormEditor/FormEditor.module.css:7]: https://github.com/taskany-inc/bricks/pull/848/files#diff-9a5a554f0b096799ebe7b4f342108145ee16a5ccdcafb3eac6289563daf8d356R1-R4

File: src/harmony/FormEditor/FormEditor.tsx [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+                    className={cn({ [s.FormEditor_disabled]: disabled }, { [s.FormEditor_filled]: value }, className)}
+                >
+                    {nullable(isDragActive && !disableAttaches, () => (
+                        <div className={s.UploadInput}>
+                            <input {...uploadInputProps} />
+                        </div>
+                    ))}
+
+                    {nullable(!focused && !value && placeholder, () => (
+                        <Text className={s.FormEditorPlaceholder} color="var(--text-ghost)" size="s" as="span">
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.tsx:299]: use css, don't use `color="var(--text-ghost)"`

File: src/harmony/FormEditor/FormEditor.module.css [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+.FormEditorWrapper.FormEditor_focused,
+.FormEditorWrapper .FormEditor_focused,
+.FormEditor_focused .EditorBackground,
+.FormEditor_filled .EditorBackground {
+    background-color: var(--input-fill-focused-borderless);
+}
+
+.FormEditorWrapper.FormEditorWrapper_view_danger.FormEditor_focused
```
Comment (awinogradov) [inline src/harmony/FormEditor/FormEditor.module.css:112]: Can u explain what is happening? How element can be on both levels?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/harmony/FormEditor/FormEditor.module.css (hunk 9 above)
Comment (DenisVorop) [reply src/harmony/FormEditor/FormEditor.module.css:122]: https://github.com/taskany-inc/bricks/pull/848/files#diff-9a5a554f0b096799ebe7b4f342108145ee16a5ccdcafb3eac6289563daf8d356R106-R114

**Review by awinogradov — APPROVED** [review]

**Pull request merged**